coordinator/gscoordinator/utils.py

```diff
@@ -443,6 +443,15 @@ def op_pre_process(op, op_result_pool, key_to_op, **kwargs):  # noqa: C901
     if op.op == types_pb2.OUTPUT:
         _pre_process_for_output_op(op, op_result_pool, key_to_op, **kwargs)
 
+def _pre_process_for_report_graph_op(op, op_result_pool, key_to_op, **kwargs):
+    assert len(op.parents) <= 1
```

Comment:
- we don't use assert, which only affected in debug mode, and not elegant exit. 
- comments.

Reply:
these code has been removed.